openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -554,6 +555,11 @@ def login_user(request, api_version='v1'):
 
         possibly_authenticated_user = user
 
+        try:
+            possibly_authenticated_user = PreLoginFilter.run(user=possibly_authenticated_user)
+        except PreLoginFilter.PreventLogin as exc:
+            raise AuthFailedError(str(exc)) from exc
```

Comment:
I'm also not clear why we're using the pattern of using `str(exc)` as the message for these exceptions. Is the idea that we'd lose the original source exception information otherwise? Or do we expect that the exception message might actually be bubbled up to users directly?

Reply:
1.  What about now? The PreLoginException is looking like [this](https://github.com/eduNEXT/openedx-filters/pull/20/files#diff-13064a3a55b3242edc585dd9d850e0278f524a2346b48f06b9202fdb08fa74f2R60-R65) and from our step, like [this](https://github.com/eduNEXT/openedx-filters-samples/blob/master/openedx_filters_samples/samples/pipeline.py#L167)

2. We -as it is right now, this can change if necessary- expect the developer to describe the error in a friendly and helpful way in the exception message to show it to the end-user. This way, we define just one helpful message, and the error information is logged in for the developer to see.  What do you think it's best? Maybe we could use a `developer_message` and `user_message` :thinking: